internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -3662,3 +3662,94 @@ resource "azurerm_kubernetes_cluster_node_pool" "pool2" {
 		data.RandomInteger,     // kubernetes_cluster dns_prefix
 	)
 }
+
+func TestAccKubernetesClusterNodePool_modeGateway(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_node_pool", "test")
+	r := KubernetesClusterNodePoolResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.modeGatewayConfig(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("mode").HasValue("Gateway"),
+				check.That(data.ResourceName).Key("gateway_public_ip_prefix_size").HasValue("30"),
```

Comment:
non-compute fields will be validated by the test framework
```suggestion
			Check: acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).ExistsInAzure(r),
```